File: lib/sys/mac_sys.c
#include <sys/select.h>

// cImport doesn't like macros, so wrap them in functions.
void sys_FD_SET(int n, fd_set* p) {
    FD_SET(n, p);
}

void sys_FD_ZERO(fd_set* p) {
    FD_ZERO(p);
}

File: cosmic/mac_sys.zig
const c = @cImport(
    @cInclude("sys/select.h"),
);

pub const timeval = c.timeval;
pub const fd_set = c.fd_set;
pub const select = c.select;

pub extern fn sys_FD_SET(n: c_int, p: *c.fd_set) void;
pub extern fn sys_FD_ZERO(p: *c.fd_set) void;